newrelic/common/signature.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from newrelic.packages import six
+
+if six.PY3:
+    from inspect import Signature
+
+    def bind_args(func, args, kwargs, unwrap=False):
```

Comment:
If I'm remembering correctly, the follow_wrapped didn't work for us with the decorators in elasticsearch. While I realize not all decorators manipulate parameters the way es did, I think it would be safer for future developers of the agent if we just don't support wrapping at all for these auto sig functions. That way future devs won't be tempted to use these functions in a potentially unsafe manor. It might also be good to capture that in a comment somewhere in this file. Perhaps in a doc string of the function?

If you feel strongly that we should keep the support for setting follow_wrapped, we absolutely should add a warning about this in the doc string for future devs. I'd argue that since we don't have a current use case for utilizing follow_wrapped though, that it would be better to simply just not support it right now and have the discussion later if we find a case where we would like to use it.